src/main/java/org/jboss/forge/addon/swarm/ui/CreateTestClassCommand.java

```diff
@@ -0,0 +1,224 @@
+              .add(asClient)
+              .add(mainClass)
+              .add(archiveType);
+   }
+
+   protected String calculateDefaultPackage(UIContext context)
+   {
+      String packageName;
+      Project project = getSelectedProject(context);
+      if (project != null)
+      {
+         packageName = project.getFacet(JavaSourceFacet.class).getBasePackage();
+      }
+      else
+      {
+         packageName = "";
+      }
+      return packageName;
+   }
+
+   @Override
+   public Result execute(UIExecutionContext context) throws Exception {
+      Project project = getSelectedProject(context);
+
+      final DependencyFacet dependencyFacet = project.getFacet(DependencyFacet.class);
+      if (!isArquillianWildflySwarmDependencyInstalled(dependencyFacet)) {
+         installArquillianWildflySwarmDependency(dependencyFacet);
+      }
+
+      JavaClassSource test = Roaster.create(JavaClassSource.class)
+              .setPackage(targetPackage.getValue())
+              .setName(named.getValue());
+
+      addArquillianRunner(test);
+      addDefaultDeploymentAnnotation(test, project);
+      addArquillianResourceEnricher(test);
+      addTestMethod(test);
+
+      JavaSourceFacet facet = project.getFacet(JavaSourceFacet.class);
+      facet.saveTestJavaSource(test);
+
+      return Results.success(String.format("Test Class %s.%s was created", targetPackage.getValue(), named.getValue()));
+   }
+
+   private void addArquillianResourceEnricher(JavaClassSource test) {
+      if (asClient.hasValue())
+      {
+         test.addImport("org.jboss.arquillian.test.api.ArquillianResource");
+         final FieldSource<JavaClassSource> urlField = test.addField();
+         urlField
+                 .setName("url")
+                 .setType(URL.class)
+                 .setPrivate();
+
+         urlField.addAnnotation("ArquillianResource");
+
+      }
+   }
+
+   private void addTestMethod(JavaClassSource test) {
+
+      MethodSource<JavaClassSource> testMethod = test.addMethod()
+              .setPublic()
+              .setReturnTypeVoid()
+              .setName("should_start_service")
+              .setBody("");
+      testMethod.addAnnotation(Test.class);
+   }
+
+   private void addArquillianRunner(JavaClassSource test) {
+      test.addAnnotation(RunWith.class).setClassValue(Arquillian.class);
+   }
+
+   private void addDefaultDeploymentAnnotation(JavaClassSource test, Project project) throws ClassNotFoundException, IOException {
+      test.addImport("org.wildfly.swarm.arquillian.DefaultDeployment");
+      final AnnotationSource<JavaClassSource> defaultDeploymentAnnotation = test.addAnnotation("DefaultDeployment");
+      if (asClient.hasValue())
+      {
+         defaultDeploymentAnnotation.setLiteralValue("testable", "false");
+      }
+
+      if (mainClass.hasValue())
+      {
+         ClassLoaderFacet facet = project.getFacet(ClassLoaderFacet.class);
+         try (URLClassLoader classLoader = facet.getClassLoader()) {
```

Comment:
Ok, now that I know the purpose of `literal` I think I can rewrite everything to avoid these dependencies.